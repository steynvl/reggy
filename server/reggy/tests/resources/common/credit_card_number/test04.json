{
  "type": "Credit card number",
  "information": {
    "visa": true,
    "dinersClub": true,
    "masterCard": true,
    "discover": true,
    "americanExpress": true,
    "jcb": true,
    "spacesAndDashes": "No spaces or dashes"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
