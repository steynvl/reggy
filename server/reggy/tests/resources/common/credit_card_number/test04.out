Java
Pattern regex = Pattern.compile("3(?:0[0-5]|[68]\\d)\\d{11}|(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\\d{12}|4\\d{12}(?:\\d{3})?|3[47]\\d{13}|(?:2131|1800|35\\d{3})\\d{11}|6(?:011|5\\d{2})\\d{12}");
3(?:0[0-5]|[68]\\d)\\d{11}|(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\\d{12}|4\\d{12}(?:\\d{3})?|3[47]\\d{13}|(?:2131|1800|35\\d{3})\\d{11}|6(?:011|5\\d{2})\\d{12}
Perl
my $regex = /6(?:011|5\d{2})\d{12}|(?:2131|1800|35\d{3})\d{11}|4\d{12}(?:\d{3})?|3[47]\d{13}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}|3(?:0[0-5]|[68]\d)\d{11}/;
6(?:011|5\d{2})\d{12}|(?:2131|1800|35\d{3})\d{11}|4\d{12}(?:\d{3})?|3[47]\d{13}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}|3(?:0[0-5]|[68]\d)\d{11}
POSIX
6(?:011|5\d{2})\d{12}|(?:2131|1800|35\d{3})\d{11}|3(?:0[0-5]|[68]\d)\d{11}|4\d{12}(?:\d{3})?|3[47]\d{13}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}
6(?:011|5\d{2})\d{12}|(?:2131|1800|35\d{3})\d{11}|3(?:0[0-5]|[68]\d)\d{11}|4\d{12}(?:\d{3})?|3[47]\d{13}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}
Python
regex = re.compile(r'3(?:0[0-5]|[68]\d)\d{11}|(?:2131|1800|35\d{3})\d{11}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}|4\d{12}(?:\d{3})?|3[47]\d{13}|6(?:011|5\d{2})\d{12}')
3(?:0[0-5]|[68]\d)\d{11}|(?:2131|1800|35\d{3})\d{11}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}|4\d{12}(?:\d{3})?|3[47]\d{13}|6(?:011|5\d{2})\d{12}
JavaScript
const regex = /3(?:0[0-5]|[68]\d)\d{11}|4\d{12}(?:\d{3})?|3[47]\d{13}|(?:2131|1800|35\d{3})\d{11}|6(?:011|5\d{2})\d{12}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}/;
3(?:0[0-5]|[68]\d)\d{11}|4\d{12}(?:\d{3})?|3[47]\d{13}|(?:2131|1800|35\d{3})\d{11}|6(?:011|5\d{2})\d{12}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}
PHP
$regex = '/3[47]\d{13}|6(?:011|5\d{2})\d{12}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}|4\d{12}(?:\d{3})?|3(?:0[0-5]|[68]\d)\d{11}|(?:2131|1800|35\d{3})\d{11}/';
3[47]\d{13}|6(?:011|5\d{2})\d{12}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}|4\d{12}(?:\d{3})?|3(?:0[0-5]|[68]\d)\d{11}|(?:2131|1800|35\d{3})\d{11}
Golang
regex, _ := regexp.Compile("4\\d{12}(?:\\d{3})?|3(?:0[0-5]|[68]\\d)\\d{11}|(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\\d{12}|6(?:011|5\\d{2})\\d{12}|(?:2131|1800|35\\d{3})\\d{11}|3[47]\\d{13}")
4\\d{12}(?:\\d{3})?|3(?:0[0-5]|[68]\\d)\\d{11}|(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\\d{12}|6(?:011|5\\d{2})\\d{12}|(?:2131|1800|35\\d{3})\\d{11}|3[47]\\d{13}
Rust
let re = Regex::new(r"6(?:011|5\d{2})\d{12}|3[47]\d{13}|3(?:0[0-5]|[68]\d)\d{11}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}|4\d{12}(?:\d{3})?|(?:2131|1800|35\d{3})\d{11}").unwrap();
6(?:011|5\d{2})\d{12}|3[47]\d{13}|3(?:0[0-5]|[68]\d)\d{11}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}|4\d{12}(?:\d{3})?|(?:2131|1800|35\d{3})\d{11}
C#
var re = new Regex(@"6(?:011|5\d{2})\d{12}|3[47]\d{13}|3(?:0[0-5]|[68]\d)\d{11}|(?:2131|1800|35\d{3})\d{11}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}|4\d{12}(?:\d{3})?");
6(?:011|5\d{2})\d{12}|3[47]\d{13}|3(?:0[0-5]|[68]\d)\d{11}|(?:2131|1800|35\d{3})\d{11}|(?:5[1-5]\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\d{12}|4\d{12}(?:\d{3})?
Scala
val re = "(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\\d{12}|4\\d{12}(?:\\d{3})?|6(?:011|5\\d{2})\\d{12}|3[47]\\d{13}|(?:2131|1800|35\\d{3})\\d{11}|3(?:0[0-5]|[68]\\d)\\d{11}".r
(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\\d{12}|4\\d{12}(?:\\d{3})?|6(?:011|5\\d{2})\\d{12}|3[47]\\d{13}|(?:2131|1800|35\\d{3})\\d{11}|3(?:0[0-5]|[68]\\d)\\d{11}
Kotlin
val regex = Regex("(?:2131|1800|35\\d{3})\\d{11}|3(?:0[0-5]|[68]\\d)\\d{11}|4\\d{12}(?:\\d{3})?|3[47]\\d{13}|6(?:011|5\\d{2})\\d{12}|(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\\d{12}")
(?:2131|1800|35\\d{3})\\d{11}|3(?:0[0-5]|[68]\\d)\\d{11}|4\\d{12}(?:\\d{3})?|3[47]\\d{13}|6(?:011|5\\d{2})\\d{12}|(?:5[1-5]\\d{2}|2720|27[01][0-9]|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9])\\d{12}